File: tests/runtimes/python/python_no_artifact.py
# Copyright Metatype OÜ, licensed under the Mozilla Public License Version 2.0.
# SPDX-License-Identifier: MPL-2.0

from typegraph.graph.typegraph import Graph
from typegraph.policy import Policy
from typegraph.runtimes.python import PythonRuntime

from typegraph import t, typegraph


@typegraph()
def python_no_artifact(g: Graph):
    public = Policy.public()
    python = PythonRuntime()

    g.expose(
        test_lambda=python.from_lambda(
            t.struct({"a": t.string()}),
            t.string(),
            lambda x: x["a"],
        ).with_policy(public),
    )

File: tests/runtimes/python/python_no_artifact.ts
// Copyright Metatype OÜ, licensed under the Mozilla Public License Version 2.0.
// SPDX-License-Identifier: MPL-2.0

import { Policy, t, typegraph } from "@typegraph/sdk";
import { PythonRuntime } from "@typegraph/sdk/runtimes/python";
import outdent from "outdent";

const tpe = t.struct({
  a: t.string(),
  b: t.list(t.either([t.integer(), t.string()])),
});

export const tg = await typegraph("python_no_artifact", (g: any) => {
  const python = new PythonRuntime();
  const pub = Policy.public();

  g.expose({
    identityLambda: python
      .fromLambda(t.struct({ input: tpe }), tpe, {
        code: "lambda x: x['input']",
      })
      .withPolicy(pub),
    identityDef: python
      .fromDef(t.struct({ input: tpe }), tpe, {
        code: outdent`
        def identity(x):
          return x['input']
        `,
      })
      .withPolicy(pub),
  });
});
